torch/csrc/autograd/python_function.h

```diff
@@ -43,6 +45,10 @@ struct PyNode : public Node {
   // THPFunction this Function is wrapping.  Owning!
   PyObject* obj;
 
+  // The AutogradCompilerCall::backwards idx corresponding to this node's
+  // backward method
+  std::optional<int> _backward_idx;
```

Comment:
Likely not needed.  Can just treat the function as a hook, and the saved variables as normal inputs.

Reply:
Wouldn't we still need indices? Something like:
```
def proxy_call_backward(...):
    ...
    backward_fn = self.hooks_proxy[backward_idx]
    saved_variables = self.args_proxy[saved_variables_idx]
```

Reply:
`saved_variables` shouldn't exist.

It should be:
```
call_backwards(hooks[0], saved_variables=[inputs[0], inputs[2], inputs[3]], ....)
```

saved_variables is just a list of tensors, which we already support in other places.